bundles/org.openhab.core/src/main/java/org/openhab/core/util/StringUtils.java

```diff
@@ -0,0 +1,318 @@
+
+        StringBuilder processed = new StringBuilder();
+        for (String splitted : str.split("\\s+")) {
+            if (splitted.length() > 1) {
+                processed.append(splitted.substring(0, 1).toUpperCase());
+                processed.append(splitted.substring(1));
+            } else {
+                processed.append(splitted.toUpperCase());
+            }
+            processed.append(" ");
+        }
+        processed.setLength(Math.max(processed.length() - 1, 0));
+        return processed.toString();
+    }
+
+    /**
+     * Pads the string from the left
+     * 
+     * <pre>
+      * padLeft("9", 4, "0")        => "0009"
+      * padLeft("3112", 12, "*")    => "********3112"
+      * padLeft("openHAB", 4, "*")  => "openHAB"
+     * </pre>
+     * 
+     * @param str the String to pad, may be null
+     * @param minSize the minimum String size to return
+     * @param padString the String to add when padding
+     * @return the padded String
+     */
+    public static String padLeft(@Nullable String str, int minSize, String padString) {
+        if (str == null) {
+            str = "";
+        }
+
+        return String.format("%" + minSize + "s", str).replace(" ", padString);
+    }
+
+    /**
+     * Creates a random string
+     * 
+     * @param length the length of the String to return
+     * @param charset the characters to use to create the String
+     * @return the random String
+     */
+    public static String randomString(int length, String charset) {
+        StringBuilder sb = new StringBuilder(length);
+        for (int i = 0; i < length; i++) {
+            int index = (int) (charset.length() * Math.random());
+            sb.append(charset.charAt(index));
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * Creates a random string with [A-Zaz] characters
+     * 
+     * @param length the length of the String to return
+     * @return the random String
+     */
+    public static String randomAlphabetic(int length) {
+        return StringUtils.randomString(length, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvxyz");
+    }
+
+    /**
+     * Creates a random string with only hex characters
+     * 
+     * @param length the length of the String to return
+     * @return the random String
+     */
+    public static String randomHex(int length) {
+        return StringUtils.randomString(length, "0123456789ABCDEF");
+    }
+
+    /**
+     * Creates a random string with [A-Za-z0-9] characters
+     * 
+     * @param length the length of the String to return
+     * @return the random String
+     */
+    public static String randomAlphanumeric(int length) {
+        return StringUtils.randomString(length, "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghijklmnopqrstuvxyz");
+    }
+
+    /**
+     * Splits the string by character type into an array
+     * 
+     * <pre>
+     * "ab de fg"   => "ab", " ", "de", " ", "fg";
+     * "ab   de fg" => "ab", " ", "de", " ", "fg";
```

Comment:
is this example correct? Did I overlook the code which changes the 3 spaces to one?

Reply:
test case indicates that example is wrong....

Reply:
Good catch, thanks